test/TEST-08-INITRD/test.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -e
+
+TEST_DESCRIPTION="Test various scenarios involving transition from/to initrd"
+TEST_NO_NSPAWN=1
+
+# shellcheck source=test/test-functions
+. "${TEST_BASE_DIR:?}/test-functions"
+
+test_append_files() {
+    local workspace="${1:?}"
+    local sd_initrd file dir
+
+    # Create a shutdown initrd
+    #
+    # This should provide coverage for shutdown initrd related issues, see:
+    #   - https://github.com/systemd/systemd/issues/28645
+    #   - https://github.com/systemd/systemd/pull/28648
+    #   - https://github.com/systemd/systemd/pull/28793
+    #
+    # This is a bit messier than I originally anticipated, as installing our own libraries
+    # is handled implicitly by install_systemd() which I don't want to use here, since
+    # I need only the systemd-shutdown binary
+    sd_initrd="$workspace/shutdown-initrd"
+    mkdir -p "$sd_initrd/etc" "$sd_initrd/usr/lib/systemd" "$sd_initrd/usr/lib64"
+    initdir="$sd_initrd" image_install bash /usr/lib/os-release
+    ln -srf "$sd_initrd/usr/lib/os-release" "$sd_initrd/etc/initrd-release"
+    cp -u "$workspace/usr/lib/systemd/systemd-shutdown" "$sd_initrd/usr/lib/systemd/systemd-shutdown"
+    initdir="$sd_initrd" inst_libs "$sd_initrd/usr/lib/systemd/systemd-shutdown"
+    # We need to deal with libsystemd stuff explicitly, as we don't call install_systemd() here
+    while read -r file; do
+        # /.../workspace/usr/lib64/systemd/foo.so => /usr/lib64/systemd
+        dir="$(dirname "${file##"$workspace"}")"
+        mkdir -p "${dir:?}"
+        cp -u "$file" "${sd_initrd:?}/${file##"$workspace"}"
```

Comment:
I think, better way is changing the line to inst_library (or inst_simple)
```
initdir="$sd_initrd" inst_library "$file" "${file##"$workspace"}"
```

Reply:
Thank you! That makes the code look like less like scrambled eggs.